hooks/tk-multi-publish2/basic/publish_movie.py

```diff
@@ -473,8 +737,144 @@ def _unreal_render_sequence_to_movie(self, destination_path, unreal_map_path, se
+        # TODO: check which codec we should use.
+
+        # We render in a forked process that we can control.
+        # It would be possible to render in from the running process using an
+        # Executor, however it seems to sometimes deadlock if we don't let Unreal
+        # process its internal events, rendering is asynchronous and being notified
+        # when the render completed does not seem to be reliable.
+        # Sample code:
+        #    exc = unreal.MoviePipelinePIEExecutor()
+        #    # If needed, we can store data in exc.user_data
+        #    # In theory we can set a callback to be notified about completion
+        #    def _on_movie_render_finished_cb(executor, result):
+        #       print("Executor %s finished with %s" % (executor, result))
+        #    # exc.on_executor_finished_delegate.add_callable(_on_movie_render_finished_cb)
+        #    r = qsub.render_queue_with_executor_instance(exc)
+
+        # We can't control the name of the manifest file, so we save and then rename the file.
+        _, manifest_path = unreal.MoviePipelineEditorLibrary.save_queue_to_manifest_file(queue)
+        manifest_path = os.path.abspath(manifest_path)
+        manifest_dir, manifest_file = os.path.split(manifest_path)
+        f, new_path = tempfile.mkstemp(
+            suffix=os.path.splitext(manifest_file)[1],
+            dir=manifest_dir
+        )
+        os.close(f)
+        os.replace(manifest_path, new_path)
+
+        self.logger.debug("Queue manifest saved in %s" % new_path)
+        # We now need a path local to the unreal project "Saved" folder.
+        manifest_path = new_path.replace(
+            "%s%s" % (
+                os.path.abspath(
+                    os.path.join(unreal.SystemLibrary.get_project_directory(), "Saved")
+                ),
+                os.path.sep,
+            ),
+            "",
+        )
+        self.logger.debug("Manifest short path: %s" % manifest_path)
+        # Command line parameters were retrieved by submitting a queue in Unreal Editor with
+        # a MoviePipelineNewProcessExecutor executor.
+        # https://docs.unrealengine.com/4.27/en-US/PythonAPI/class/MoviePipelineNewProcessExecutor.html?highlight=executor
+        cmd_args = [
+            sys.executable,
+            "%s" % os.path.join(
+                unreal.SystemLibrary.get_project_directory(),
+                "%s.uproject" % unreal.SystemLibrary.get_game_name(),
+            ),
+            "MoviePipelineEntryMap?game=/Script/MovieRenderPipelineCore.MoviePipelineGameMode",
+            "-game",
+            "-Multiprocess",
+            "-NoLoadingScreen",
+            "-FixedSeed",
+            "-log",
+            "-Unattended",
+            "-messaging",
+            "-SessionName=\"Publish2 Movie Render\"",
+            "-nohmd",
+            "-windowed",
+            "-ResX=1280",
+            "-ResY=720",
+            # TODO: check what these settings are
```

Comment:
Should these come from the game values itself or always be statically overwritten??